src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -278,6 +278,28 @@ public void SetAllExceptions(enum_EXCEPTION_STATE dwState)
             }
         }
 
+        public int TryGetExceptionBreakpoint(string bkptno, out EXCEPTION_INFO exceptionInfo)
+        {
+            const string CppExceptionCategoryString = "{3A12D0B7-C26C-11D0-B442-00A0244A1DD2}";
+            exceptionInfo = new EXCEPTION_INFO();
+            ExceptionCategorySettings categorySettings;
+            if (_categoryMap.TryGetValue(new Guid(CppExceptionCategoryString), out categorySettings))
+            {
+                ulong breakpointNumber = Convert.ToUInt32(bkptno, CultureInfo.InvariantCulture);
+                if (categorySettings.CurrentRules.ContainsValue(breakpointNumber))
+                {
+                    // test -- need to delete; these are placeholder values
+                    exceptionInfo.bstrExceptionName = categorySettings.CategoryName;
+                    exceptionInfo.dwCode = (uint)breakpointNumber;
+                    bool isBreakThrown = categorySettings.CategoryState == ExceptionBreakpointStates.BreakThrown; // this is true, but why is dwState still NONE?
+                    exceptionInfo.dwState = isBreakThrown ? (enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE) : enum_EXCEPTION_STATE.EXCEPTION_NONE;
```

Comment:
>enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE [](http://example.com/codeflow?start=61&length=104)

You can just always use this value (isBreakThrown will always be true)

Reply:
> isBreakThrown will always be true

Makes sense when reason == "breakpoint-hit". That said, I can just remove the `exceptionInfo.dwState` assignment then.

Instead of the following in DebuggedProcess.cs, Line 1194
```
ExceptionBreakpointStates exceptionBreakpointStates = ExceptionManager.ToExceptionBreakpointState(exceptionInfo.dwState);
_callback.OnException(thread, exceptionInfo.bstrExceptionName, "", 0, exceptionInfo.guidType, exceptionBreakpointStates);
```

I can simply use `_callback.OnException(thread, exceptionInfo.bstrExceptionName, "", 0, exceptionInfo.guidType, ExceptionBreakpointStates.BreakThrown);`

Reply:
Yup. Though in that case I would suggest not using `EXCEPTION_INFO`, and instead just return the guid+name from this method.